pkg/apis/flowcontrol/types.go

```diff
@@ -88,10 +88,11 @@ type FlowSchemaSpec struct {
 	PriorityLevelConfiguration PriorityLevelConfigurationReference
 	// `matchingPrecedence` is used to choose among the FlowSchemas that match a given request. The chosen
 	// FlowSchema is among those with the numerically lowest (which we take to be logically highest)
-	// MatchingPrecedence.  Each MatchingPrecedence value must be non-negative.
+	// MatchingPrecedence.  Each MatchingPrecedence value must be ranged in [0,10000]. For two FlowSchemas
+	// with conflicting MatchingPrecedence, the one created earlier is prior.
 	// Note that if the precedence is not specified or zero, it will be set to 1000 as default.
```

Comment:
maybe this will become clear below, but if not specified and zero are treated the same, why make this a pointer?

Reply:
i updated the godoc for clarification. now it's "if not specified ~~and zero~~"

> why make this a pointer?

https://github.com/kubernetes/kubernetes/pull/85268#discussion_r352905593 and `0` sounds like a better "highest" than `1` in some sense

Reply:
Personally I would have made 0 the logically highest priority and made this field a pointer, but I don't think it's important enough to ask for a change.